monitors.go

```diff
@@ -66,6 +66,7 @@ type Monitor struct {
 	Query   *string  `json:"query,omitempty"`
 	Name    *string  `json:"name,omitempty"`
 	Message *string  `json:"message,omitempty"`
+	State   *string  `json:"overall_state,omitempty"`
```

Comment:
When checking the DD API docs at https://docs.datadoghq.com/api/?lang=bash#monitors I noticed both "state" and "overal_state" exist, and have different meanings. Unmarshalling "overall_state" into "State" would be confusing.

Reply:
Completely missed regular state, thanks! I've updated this to be inline with the json.